### PR TITLE
Built common CommandsBase so we can share double-dashed options from the...

### PR DESCRIPTION
**Pull request by ed-**
... parser between command classes.

Double-dashed options can happen in any order, even interspersed with the args.

Try 

``` bash
reddwarf-cli instance --id=$ID get
```

The resource (instance/database/user/) and action (get/list/create/delete) must still come in that order relative only to each other. optparse pulls them out in that order.

### CODE REVIEW

File: reddwarfclient/cli.py [hunk 1]
```diff
@@ -316,25 +257,11 @@ def main():
         # Check for a valid action and perform that action
         action = args.pop(0)
         if action in actions:
-            fn = actions.get(action)
-
+            #fn = actions.get(action)
```
Comment (TimSimpsonR) [inline reddwarfclient/cli.py]: Cruft?

File: reddwarfclient/cli.py [hunk 2]
```diff
@@ -305,7 +245,8 @@ def main():
     cmd = args.pop(0)
     if cmd in COMMANDS:
         fn = COMMANDS.get(cmd)
-        command_object = fn()
+        command_object = fn(oparser)
+        #command_object.parse_options(oparser)
```
Comment (TimSimpsonR) [inline reddwarfclient/cli.py]: Cruft?

**Comment by TimSimpsonR** [comment]
+1

File: reddwarfclient/cli.py (hunk 1 above)
Comment (ed-) [inline reddwarfclient/cli.py]: # YOU'RE CRUFT

**Comment by jcru** [comment]
+1  Very nice!

**Comment by pdmars** [comment]
+1

**Pull request merged**